app/contracts/projects/base_contract.rb

```diff
@@ -75,6 +75,14 @@ def available_custom_fields
       end
     end
 
+    def all_available_custom_fields
+      if user.admin?
+        model.all_available_custom_fields
+      else
+        model.all_available_custom_fields.reject(&:admin_only?)
```

Comment:
I'm wondering if it is better to use `reject`, a `where` or `visible(project: model)`

Also `.reject(&:admin_only?)` in `available_custom_fields` is probably useless as `Projec#available_custom_fields` already checks that

Reply:
Since this contract is also being used for the API, we cannot apply the visibility checks here. The API's behaviour is to allow writing custom field values the user has access to, even if the fields are not enabled in the project, so the only check we can do is the admin flag.

Maybe we could use a `where` condition here, however for the sake of uniformity with the `available_custom_fields` I prefer using the `reject` method.

Reply:
If API uses not only `all_available_custom_fields`, but also `available_custom_fields`, it still uses visibility check. Quickly searching shows that method is present in API code, but no idea if any of those involve `Project#available_custom_fields`